ethcore/src/executive.rs

```diff
@@ -276,7 +276,7 @@ impl<'a, B: 'a + StateBackend> Executive<'a, B> {
 
 			let cost = builtin.cost(data);
 			if cost <= params.gas {
-				builtin.execute(data, &mut output);
+				builtin.execute(data, &mut output)?;
```

Comment:
I think every other exit from this function is either doing: `state.discard_snapshot()`, `state.revert_to_snapshot()` or `enact_result` (which does this stuff internally afair).

This could exit the function after creating `state.checkpoint()` but without reseting it. Also seems that tracing for top-level failed builtin call might be missing.